lib/src/widget/nav_bar/widget/nav_bar_container.dart

```diff
@@ -0,0 +1,159 @@
+import 'package:flutter/material.dart';
+import 'package:xayn_design/src/widget/nav_bar/data/config_pair.dart';
+import 'package:xayn_design/src/widget/nav_bar/widget/nav_bar.dart';
+import 'package:xayn_design/xayn_design.dart';
+
+class NavBarContainer extends StatefulWidget {
+  final MaterialApp child;
+
+  const NavBarContainer({
+    Key? key,
+    required this.child,
+  }) : super(key: key);
+
+  @override
+  _NavBarContainerState createState() => _NavBarContainerState();
+
+  /// Call this, when you need to update current instance of the [NavBar]
+  /// f.e. [NavBarConfig] was changed for the screen
+  static void updateNavBar(BuildContext context) =>
+      context.dependOnInheritedWidgetOfExactType<_Updater>()!.data.updateBar();
```

Comment:
❗ instead of usin `!` an assert with a nice message would be good so the user knows why this fails (because we couldn't find a NAvBar)

Reply:
fixed